app/models/site_status.rb

```diff
@@ -43,4 +46,27 @@ class SiteStatus < ApplicationRecord
   }
   scope :with_vacancies, -> { where.not(vac_status: :no_vacancies) }
   scope :open_for_applications, -> { findable.applications_being_accepted_now.with_vacancies }
+
+private
+
+  def vac_status_must_be_consistent_with_course_study_mode
+    unless vac_status_consistent_with_course_study_mode?
+      errors.add(:vac_status, "(#{vac_status}) must be consistent with course study mode #{course.study_mode}")
```

Comment:
Think it's needs message

errors.add(:vac_status,
                 :invalid,
                 message:

Reply:
As it is works fine, but you highlighted that the tests don't check for the validation error, which is really important!